Sources/Sentry/SentryBaggage.m

```diff
@@ -0,0 +1,59 @@
+#import "SentryBaggage.h"
+#import "SentryDsn.h"
+#import "SentryLog.h"
+#import "SentryOptions+Private.h"
+#import "SentryScope+Private.h"
+#import "SentrySerialization.h"
+#import "SentryTraceContext.h"
+#import "SentryTracer.h"
+#import "SentryUser.h"
+
+@implementation SentryBaggage
+
+- (instancetype)initWithTraceId:(SentryId *)traceId
+                      publicKey:(NSString *)publicKey
+                    releaseName:(nullable NSString *)releaseName
+                    environment:(nullable NSString *)environment
+                    transaction:(nullable NSString *)transaction
+                         userId:(nullable NSString *)userId
+                    userSegment:(nullable NSString *)userSegment
+{
+
+    if (self = [super init]) {
+        _traceId = traceId;
+        _publicKey = publicKey;
+        _releaseName = releaseName;
+        _environment = environment;
+        _transaction = transaction;
+        _userId = userId;
+        _userSegment = userSegment;
+    }
+
+    return self;
+}
+
+- (nullable NSString *)toHTTPHeader
+{
+    NSMutableDictionary *information =
```

Comment:
We should limit the total header value length to 8192 characters as per https://www.w3.org/TR/baggage/#limits

> A platform MUST propagate all list-members including any list-members added by the platform if the resulting baggage-string would be 8192 bytes or less. If the resulting baggage-string would be greater than 8192 bytes, some list-members MAY be dropped until the resulting baggage-string is 8192 characters or less.

Guess this implementation doesn't have to worry about the 64 list-member limit since we only add our own and nothing else.

Reply:
To be honest, for mobile I don't think we even need to worry about the size limit. But I will include it.